synapse/federation/federation_client.py

```diff
@@ -272,7 +272,7 @@ async def backfill(
 
         # Check signatures and hash of pdus, removing any from the list that fail checks
         pdus[:] = await self._check_sigs_and_hash_and_fetch(
-            dest, pdus, outlier=True, room_version=room_version
+            dest, pdus, room_version=room_version
```

Comment:
note that the events returned by `event_from_pdu_json` a couple of lines above are *not* flagged as outliers. the `outlier` flag here tells `_check_sigs_and_hash_and_fetch` that, if it pulls another copy of the event due to an incorrect signature, it should flag it as an outlier.

(I think the existence of an `outlier` param on `_check_sigs_and_hash_and_fetch` is misguided, but that's a separate PR.)